docs/manual/scala/guide/build/ScalaComponents.md

```diff
@@ -0,0 +1,68 @@
+
+This is a brief introduction to Lagom Application cake, there's more information in [[Wiring together a Lagom application|ServiceImplementation#Wiring-together-a-Lagom-application]].
+
+## Components
+
+This is a list of available `Components` you may use to build your application cake. This list contains Components provided by Lagom, Play and other projects or libraries. You may develop your own Components and use those instead of the provided on this list as long as the contracts are fulfilled.
+
+|  Service Components  ||
+| -------------------- | ----------- |
+| [LagomServerComponents](api/com/lightbend/lagom/scaladsl/server/LagomServerComponents.html) | a main Component for any Lagom Service. See [[Building a cake step by step|ScalaComponents#Building-a-cake-step-by-step]].|
+| [LagomServiceClientComponents](api/com/lightbend/lagom/scaladsl/client/LagomServiceClientComponents.html) | a main Component for any Lagom Service or application consuming Lagom Services. See [[Building a cake step by step|ScalaComponents#Building-a-cake-step-by-step]] and  [[Binding a service client|ServiceClients#Binding-a-service-client]].|
+| [MetricsServiceComponents](api/com/lightbend/lagom/scaladsl/server/status/MetricsServiceComponents.html) | adds a `MetricsService` to your service so you can remotely track the status of the CircuitBreakers on your service. Using this only makes sense when you Application is consuming other services (hence using remote calls protected with Circuit Breakers). See [[Circuit Breaker Metrics|ServiceClients#Circuit-breaker-metrics]] |
+
+#### Building a cake step by step
+
+`LagomServerComponents` and `LagomServiceClientComponents` are a bit special because they are already mixed into `LagomApplication` and `LagomClientApplication` respectively so you will rarely need to explicitly mix either of them into your Application. The reason why `LagomServerComponents` and `LagomServiceClientComponents` exist is so that you can build your Application cake as the sum of middle-sized portions. Instead of creating a single `abstract class` where you mix in all the Componenets, you may create one trait `MyPartialFooTrait` where you mix in `LagomServerComponents` and parts of your logic and then you could create `MyPartialBarTrait` with some other parts. Then you would build a `MyApplication` mixing in `MyPartialFooTrait` and `MyPartialBarTrait`. so far it's not very different from having a single Application where all Components are mixed. The key advantage of partially creating your cake is that you can use `MyPartialTrait` in isolation in your tests (instead of using a complete `MyApplication`) which will be lighter and provide decoupling.
```

Comment:
This paragraph seems strangely stuck in the middle of the reference tables. I think it could be moved up and titled something like "Defining your own components traits". It should start with the "why" — what is the problem to be solved — and then continue on to the "how".

I would recommend naming examples with the `Components` suffix rather than `Trait`, since that's the established convention.

I'm also not a big fan of using "Foo" and "Bar" as example names, as it can make it difficult to comprehend what a realistic example might be. Better to elaborate a real-world use case for the technique and use meaningful names in that context.